napari/_qt/qt_viewer.py

```diff
@@ -290,15 +291,19 @@ def _on_cursor(self, event):
 
     def _on_reset_view(self, event):
         if isinstance(self.view.camera, ArcballCamera):
-            quat = self.view.camera._quaternion.create_from_axis_angle(
-                *event.quaternion
+            quat = self.view.camera._quaternion.create_from_euler_angles(
+                *self.viewer.camera.angles, degrees=True,
             )
             self.view.camera._quaternion = quat
-            self.view.camera.center = event.center
-            self.view.camera.scale_factor = event.scale_factor
+            self.view.camera.center = self.viewer.camera.center[::-1]
+            self.view.camera.scale_factor = self.viewer.camera.scale
         else:
             # Assumes default camera has the same properties as PanZoomCamera
-            self.view.camera.rect = event.rect
+            corner = np.subtract(
+                self.viewer.camera.center, self.viewer.camera.scale / 2
+            )
+            rectangle = tuple(corner[::-1]) + (self.viewer.camera.scale,) * 2
+            self.view.camera.rect = rectangle
```

Comment:
What's going on in the above lines? It's pretty complicated logic. Why are coordinates now being inverted that weren't before?

Reply:
These are conversions from numpy to vispy ordering, i've added a comment.